src/connector/src/parser/avro/util.rs

```diff
@@ -281,32 +281,12 @@ pub(crate) fn from_avro_value(value: Value, value_schema: &Schema) -> Result<Dat
                 RwError::from(InternalError(err_msg))
             })?,
         ),
-        Value::TimestampMillis(millis) => ScalarImpl::NaiveDateTime(
-            NaiveDateTimeWrapper::with_secs_nsecs(
-                millis / 1_000,
-                (millis % 1_000) as u32 * 1_000_000,
-            )
-            .map_err(|e| {
-                let err_msg = format!(
-                    "avro parse error.wrong timestamp millis value {}, err {:?}",
-                    millis, e
-                );
+        Value::TimestampMicros(t) | Value::TimestampMillis(t) => {
+            ScalarImpl::Int64(i64_to_timestamptz(t).map_err(|e| {
```

Comment:
`i64_to_timestamptz` is used to guess the unit when it is not clear from the context. At this place we know it is micro or milli:
```
Value::TimestampMicros(us) => ScalarImpl::Int64(us),
Value::TimestampMillis(ms) => ScalarImpl::Int64(ms.checked_mul(1000).ok_or_else(|| ...)?),
```

The unit test below is actually an example of misuse: `1620000000000 us` = `1620000000 ms` should be `1970-01-19 18:00:00Z`, which is outside out the valid guess range. `i64_to_timestamptz` guessed the unit to be ms / s, and returned `1620000000000000 us` which is `2021-05-03 00:00:00Z`.